File: src/test/java/org/dedda/bratwurst/parse/PopParserTest.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.BratwurtstTestcase;
import org.dedda.bratwurst.lang.Pop;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
public class PopParserTest extends BratwurtstTestcase {

    @Test
    public void testParse() throws Exception {
        String variableName = "testPopVar";
        String line = "<" + variableName + "<";
        int lineNumber = 15;
        PopParser parser = new PopParser();
        Pop pop = parser.parse(line, lineNumber);
        assertEquals(variableName, pop.getVariableName());
        assertEquals(lineNumber, pop.getLineNumber());
    }
}

File: src/test/java/org/dedda/bratwurst/parse/PopParserTest.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.BratwurtstTestcase
import org.testng.Assert
import org.testng.annotations.Test

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
class PopParserTest : BratwurtstTestcase() {
    @Test
    @Throws(Exception::class)
    fun testParse() {
        val variableName = "testPopVar"
        val line = "<$variableName<"
        val lineNumber = 15
        val parser = PopParser()
        val pop = parser.parse(line, lineNumber)
        Assert.assertEquals(variableName, pop!!.variableName)
        Assert.assertEquals(lineNumber, pop.lineNumber)
    }
}